Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -336,70 +331,81 @@ - (DisplayableText *)displayableTextForInteractionId:(NSString *)interactionId
     return displayableText;
 }
 
+- (nullable TSAttachment *)firstAttachmentIfAnyOfMessage:(TSMessage *)message
+{
+    if (message.attachmentIds.count == 0) {
+        return nil;
+    }
+    NSString *_Nullable attachmentId = message.attachmentIds.firstObject;
+    if (attachmentId.length == 0) {
+        return nil;
+    }
+    return [TSAttachment fetchObjectWithUniqueID:attachmentId];
```

Comment:
This is kind of a sneaky spot to create a transaction. 

And more generally, since the ViewItem is driving UI code, it seems like we should be doing any reads from a transaction parameter passed in from a longLivedRead db connection which hangs on the view controller (e.g. `uiDatabaseConnection`)

This problem isn't new to this PR, but it just occurred to me.

Reply:
Good catch - card made.